packages/itwinui-css/src/button/variant.scss

```
@@ -1,44 +1,27 @@
 // Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 // See LICENSE.md in the project root for license terms and full copyright notice.
 
-/// @arg $variant Can be one of: cta, high-visibility
-@mixin iui-button-variant($variant) {
-  @if $variant == cta {
-    --_iui-button-background-color: var(--iui-color-background-positive);
-
-    &:hover,
-    &:active {
-      --_iui-button-background-color: var(--iui-color-background-positive-hover);
-    }
+@mixin iui-button-variant {
+  &:where([data-iui-variant='high-visibility'], [data-iui-variant='cta']) {
+    --_iui-field-color-text: var(--iui-color-white);
+    --_iui-field-color-text-hover: var(--iui-color-white);
+    --_iui-field-color-focus-outline: var(--iui-color-white);
+    --_iui-field-focus-outline-offset: -3px;
+    --_iui-field-focus-outline-width: 1px;
+    --_iui-button-icon-fill: var(--iui-color-white);
   }
 
-  @if $variant == high-visibility {
-    --_iui-button-background-color: var(--iui-color-background-accent);
-
-    &:hover,
-    &:active {
-      --_iui-button-background-color: var(--iui-color-background-accent-hover);
+  $variant-mappings: (
+    'high-visibility': 'informational',
+    'cta': 'positive',
+  );
+
+  @each $newVariant, $originalVariant in $variant-mappings {
+    &:where([data-iui-variant='#{$newVariant}']) {
+      --_iui-field-color-background: var(--iui-color-background-#{$originalVariant});
+      --_iui-field-color-background-hover: var(--iui-color-background-#{$originalVariant}-hover);
+      --_iui-field-color-border: var(--iui-color-background-#{$originalVariant});
+      --_iui-field-color-border-hover: var(--iui-color-background-#{$originalVariant}-hover);
     }
   }
```

1. prefer duplicating rather than looping for clarity
2. prefer `accent` over `informational` for high-visbility button

```suggestion
  &:where([data-iui-variant='high-visibility']) {
    --_iui-field-color-background: var(--iui-color-background-accent);
    --_iui-field-color-background-hover: var(--iui-color-background-accent-hover);
    --_iui-field-color-border: var(--iui-color-background-accent);
    --_iui-field-color-border-hover: var(--iui-color-background-accent-hover);
  }
  
  &:where([data-iui-variant='cta']) {
    --_iui-field-color-background: var(--iui-color-background-positive);
    --_iui-field-color-background-hover: var(--iui-color-background-positive-hover);
    --_iui-field-color-border: var(--iui-color-background-positive);
    --_iui-field-color-border-hover: var(--iui-color-background-positive-hover);
  }
```